### DISCUSSION TITLE
Support Center Icons

### DISCUSSION THREAD

**Discussion by HartiganHM** [twilio-labs/paste#2617] [Request: Icon] [answered]
Hey, all!

We're currently in the process of migrating the Twilio Support application over to the Console monorepo and need some new icons. The ones that are currently being used are from an old `@twiliointernal` dependency which we'd love to deprecate. The icons we need are in this screenshot for reference and are listed below.

I know that previously icon requests were on pause as Paste stability was being prioritized, so if there's anyway to get approval on these icons and then receive an SVG that we can use in our app directly (similar to a previous icon request I made) that would really help us expedite our migration.

Please let me know if you have any questions and thanks for the help!

___

**Support screenshot**
![Screen Shot 2022-08-17 at 1 10 30 PM](https://user-images.githubusercontent.com/24458700/185419228-5c425351-6aa5-4050-bb71-1b1e61a02f4d.png)

**Help documentation icon**
- Type: UI Icon
- [Option 1](https://www.streamlinehq.com/icons/streamline-mini-line/interface-essential/content/interface-content-book-1)
- [Option 2](https://www.streamlinehq.com/icons/streamline-mini-line/interface-essential/content/interface-content-book-2)

**Ticket history icon**
- Type: UI Icon
- [Option 1](https://www.streamlinehq.com/icons/streamline-mini-line/interface-essential/time/interface-time-rewind)
- [Option 2](https://www.streamlinehq.com/icons/streamline-mini-line/interface-essential/time/interface-time-hour-glass)

**Submit ticket icon**
- Type: UI Icon
- [Option 1](https://www.streamlinehq.com/icons/streamline-mini-line/entertainment/tickets/entertainment-ticket)
- [Option 2](https://www.streamlinehq.com/icons/streamline-mini-line/mail/send/mail-send-email)

**Community icon**
- Type: UI Icon
- [Option 1](https://www.streamlinehq.com/icons/streamline-mini-line/interface-essential/user/interface-user-multiple)

**Comment by serifluous** [comment]
Thanks, @HartiganHM ! We're actually working on getting some icons in our current release (including BusinessIcon), so I'll let you know when these are available in Paste, likely over the next 2 weeks.

For the ticket history icon, would the HistoryIcon we have already work for you? https://paste.twilio.design/components/icons

~~And for the community icon, we already have that glyph representing "users" in the admin center and "teams" in Flex. I'll take a look through Streamline and see if there's another glyph that can work and run it by you.~~ See below

**Reply by serifluous** [reply]
For Docs and Community, I'm actually going to grab from what's on twilio.com:
<img width="708" alt="Screen Shot 2022-08-18 at 10 26 45 AM" src="https://user-images.githubusercontent.com/3821840/185457581-33a4c839-e80b-4044-a36b-b8a7ae907c6e.png">

**Reply by HartiganHM** [reply]
Hey, @serifluous! Thanks for the quick follow-up. That's great to hear about the upcoming release. My mistake! I missed both the `History` and `ProductAdminUsers` icons. Both of those will work, but I'm open to suggestions for what you might recommend.

**Reply by serifluous** [reply]
I'll add CommunityIcon, so you get a bit of a different glyph/image than ProductAdminUsers! It'll match what's on twilio.com (in the screenshot I shared above).

For the ticket icon, what are your thoughts on using our ProductBillingIcon for "Support Plans" instead, and using SupportIcon for "Submit a Ticket"? A ticket icon doesn't seem to be so commonly used, and a more generic support icon could work for "Submit a Ticket" in case that language ever changes to "Ask for help", "Contact support", etc., especially if it needs to be translated.

Vvvv quick mockup here:
![image](https://user-images.githubusercontent.com/3821840/185489521-f9727491-2f95-4a8d-ab76-69480dcb0fc7.png)

**Reply by HartiganHM** [reply]
Ah, I really like the "Docs" and "Community" icons, those will be great. I think the alternatives for "Support Plans" and "Submit a Ticket" make sense as well, especially given the language example you mention, so I'm on board with those. We do have a designer on our team now, Eileen Tong, who I'd like to get a quick thumbs-up from, but she's in the process of creating a Twilio GH account. I'll make sure we follow-up tomorrow AM.

Thanks for the help!

**Reply by tonge495** [reply]
Hey!  Thank you for looping me in @HartiganHM!  Just catching up on the thread now.

@serifluous Thanks for the response on this!  Logic for using the documentation and community icons make sense to me, as it sounds like it will be consistent with what twilio.com already has.  Submit a ticket makes sense as well given the language example.

The only caveat for the support plan icon is that we have 3 paid support plans and one free one but the money icon would imply that all of them are paid.  Is there perhaps another icon that would indicate 'plans'?  Maybe something with a checklist?

**Reply by serifluous** [reply]
Sure! The money icon is actually our Billing Icon in Console, which I think could be ok in this case, even if one of the plans are free.

If you do want to use another icon, though, feel free to take a look through Streamline: https://www.streamlinehq.com/icons/streamline-regular and let us know if there are icons that you feel like would work here (similar to how Hugh linked to a few options in the original post).

**Reply by tonge495** [reply]
Oh got it!  Do you happen to know where else the Billing Icon is used on Console (I unfortunately don't have access, so am unable to check myself at the moment).  I think if this icon is consistent with how it's used in other parts of Console, let's just keep the Billing Icon for now.  Probably not worth introducing a new icon if it's already being used elsewhere for a similar purpose.

**Reply by serifluous** [reply]
For sure! It's mainly being used in the sidebar in the Billing section of the Console, and on overview pages for different products to link off to marketing and pricing pages. The links in the screenshot below link off to these 2 pages, as an example: https://www.twilio.com/live and https://www.twilio.com/sms/pricing/us

<img width="1658" alt="Screen Shot 2022-08-19 at 12 14 13 PM" src="https://user-images.githubusercontent.com/3821840/185691331-05686c93-7aae-4a18-b0e8-73000672e282.png">
<img width="1661" alt="Screen Shot 2022-08-19 at 12 13 40 PM" src="https://user-images.githubusercontent.com/3821840/185691372-93673f8a-96de-4648-905e-7cb65f0f84ad.png">
<img width="1660" alt="Screen Shot 2022-08-19 at 12 16 08 PM" src="https://user-images.githubusercontent.com/3821840/185691377-d087fdb9-57d7-4902-877b-aed1a101b969.png">

**Reply by tonge495** [reply]
OK, I think in that case I think the existing usage is close enough that we can just use the Billing Icon then, especially since the pricing screenshot on the 2nd screenshot also includes 'free plans.'  I'm good with this.

Thank you for checking for me!

**Reply by serifluous** [reply]
Awesome, thanks! I'll follow up here once we've got all the new icons in Paste.

**Comment by SiTaggart** [comment, marked as answer]
Hi folks

A big icon drop just landed in Figma and NPM which should cover all these use cases. Documentation and Community as an example are now available. Let us know if we missed any.

**Reply by HartiganHM** [reply]
Looks good on my end, thanks all!

**Reply by tonge495** [reply]
Thanks everyone!

**Reply by HartiganHM** [reply]
Hey, @SiTaggart, quick check here, just to see if it's on my end or a Paste update. I've updated to `9.1.0` which has all of the icons, but I'm also getting some TS errors that all icons now require a `disabled` prop. We may have missed when this was introduced, but I also don't see it as a required prop in the [docs](https://paste.twilio.design/components/icons/usage-guidelines#props), so I wanted to see what the gap might be here.

Thanks!

**Reply by HartiganHM** [reply]
**UPDATE**

Leaving this for posterity, but it looks like this was just a TS inconsistency and all icons are requiring `disabled`.

___

New little find, just to provide some more evidence, this seems to only happen with some icons. In this screenshot, the `AgentIcon` won't accept `disabled` as a prop whereas the `CommunityIcon` requires it.

![image](https://user-images.githubusercontent.com/24458700/190813297-738da74e-af04-4d70-8125-16a32c498b96.png)

**Reply by SiTaggart** [reply]
To round out this thread, offline we discovered it was a problem with Core being on v10, and needing to be on v15 for this type bug to disappear.
